src/Bindings/LuaStateParams.h

```diff
@@ -0,0 +1,571 @@
+// LuaStateParams.h
+
+// Declares the cLuaStateParams class that provides high-level functions for reading API parameters from Lua state
+
+#pragma once
+
+
+
+
+
+#include "LuaState.h"
+
+
+
+
+
+namespace Detail
+{
+	/** Utility struct that provides the functionality to cLuaStateParams::GetTypeDescription.
+	Supports decorators such as cLuaStateParams::cSelf<T>.
+	Must be a struct in order to support decorators, and must not be inside a class (gcc / clang complain about that).
+	The general declaration must not be used by the code, all types need a specialization returning the correct type. */
+	template <typename T>
+	struct TypeDescription
+	{
+		static const char * desc()
+		{
+			static_assert(false, "Type is missing a TypeDescription<> template");
```

Comment:
I think clang and gcc don't like this.  Maybe leave `desc` out altogether or even leave the struct undefined.

Reply:
Yes, I'll leave it out. I just wanted some place where the compiler would error out and there would be a direct comment about what the programmer should do if they ever encounter it. Suppose it could be put directly into the `cLuaStateParams::GetTypeDesc` function.